backend/src/gpu.py

```diff
@@ -63,7 +63,9 @@ def __init__(self):
                     uuid=nv.nvmlDeviceGetUUID(handle),
                     index=i,
                     handle=handle,  # type: ignore
-                    arch=nv.nvmlDeviceGetArchitecture(handle),
+                    arch=nv.nvmlDeviceGetArchitecture(handle)
+                    if getattr(nv, "nvmlDeviceGetArchitecture", None) is not None
+                    else nv.NVML_DEVICE_ARCH_UNKNOWN,
```

Comment:
This doesn't fix the error the user had. The error was

```
chaiNNer\python\python\Lib\site-packages\pynvml\nvml.py", line 4163, in nvmlDeviceGetArchitecture

pynvml.nvml.NVMLError_FunctionNotFound: Function Not Found
```

The error occurs inside `nvmlDeviceGetArchitecture` and here's the code of that function:

```py
def nvmlDeviceGetArchitecture(device):
    arch = _nvmlDeviceArchitecture_t()
    fn = _nvmlGetFunctionPointer("nvmlDeviceGetArchitecture")
    ret = fn(device, byref(arch))
    _nvmlCheckReturn(ret)
    return arch.value
```

The problem isn't that the *python* function `nvmlDeviceGetArchitecture` doesn't exist, but that the *C* API can't find the function `nvmlDeviceGetArchitecture` is whatever DLL from the driver nvml loads.

Reply:
oh